### PR TITLE
Add fees and estimation of tx's gas

### PR DESCRIPTION
**Pull request by NicolasMahe**
closes https://github.com/mesg-foundation/engine/issues/1564

### CODE REVIEW

File: cosmos/appfactory.go [hunk 1]
```diff
@@ -32,7 +32,7 @@ func NewAppFactory(logger log.Logger, db dbm.DB) *AppFactory {
 	cosmoscodec.RegisterCrypto(codec.Codec)
 
 	return &AppFactory{
-		baseApp: bam.NewBaseApp("engine", logger, db, auth.DefaultTxDecoder(codec.Codec)),
+		baseApp: bam.NewBaseApp("engine", logger, db, auth.DefaultTxDecoder(codec.Codec), baseapp.SetMinGasPrices("1.0mesg")),
```
Comment (NicolasMahe) [inline cosmos/appfactory.go]: `1.0mesg` should be move to config

File: cosmos/client.go [hunk 2]
```diff
@@ -86,7 +87,21 @@ func (c *Client) BuildAndBroadcastMsg(msg sdktypes.Msg, accName, accPassword str
...
+	if txBuilder.SimulateAndExecute() {
+		txBytes, err := txBuilder.BuildTxForSim([]sdktypes.Msg{msg})
+		if err != nil {
+			return nil, err
+		}
+		_, adjusted, err := authutils.CalculateGas(c.QueryWithData, codec.Codec, txBytes, txBuilder.GasAdjustment())
+		if err != nil {
+			return nil, err
+		}
+		fmt.Println("adjusted", adjusted)
```
Comment (NicolasMahe) [inline cosmos/client.go]:
```suggestion
```

File: cosmos/client.go [hunk 3]
```diff
@@ -86,7 +87,21 @@ func (c *Client) BuildAndBroadcastMsg(msg sdktypes.Msg, accName, accPassword str
...
+		txBuilder = txBuilder.WithGas(adjusted)
+	}
+
 	// TODO: cannot sign 2 tx at the same time. Maybe keybase cannot be access at the same time. Add a lock?
+	fmt.Println("txBuilder", txBuilder.Gas())
```
Comment (NicolasMahe) [inline cosmos/client.go]:
```suggestion
```

File: cosmos/genesis.go [hunk 4]
```diff
@@ -111,7 +111,8 @@ func genGenesisAppState(defaultGenesisŚtate map[string]json.RawMessage, signedS
 	genAccs := []genaccounts.GenesisAccount{}
 	for _, signer := range signedStdTx.GetSigners() {
 		stakes := sdktypes.NewCoin(sdktypes.DefaultBondDenom, sdktypes.NewInt(100000000))
-		genAcc := genaccounts.NewGenesisAccountRaw(signer, sdktypes.NewCoins(stakes), sdktypes.NewCoins(), 0, 0, "", "")
+		mesg := sdktypes.NewCoin("mesg", sdktypes.NewInt(1000000000000000))
```
Comment (NicolasMahe) [inline cosmos/genesis.go]: This should be done better. But maybe it's useless as PR https://github.com/mesg-foundation/engine/pull/1552 could be use to generate the genesis

File: cosmos/txbuilder.go [hunk 5]
```diff
@@ -17,22 +17,30 @@ type TxBuilder struct {
...
 			accSeq,
-			1000000,
+			flags.DefaultGasLimit,
 			flags.DefaultGasAdjustment,
 			true,
 			chainID,
 			"",
 			sdktypes.NewCoins(),
-			sdktypes.DecCoins{},
+			sdktypes.NewDecCoins(sdktypes.NewCoins(sdktypes.NewCoin("mesg", sdktypes.NewInt(1)))),
```
Comment (NicolasMahe) [inline cosmos/txbuilder.go]: Should be put in config

File: cosmos/appfactory.go (hunk 1 above)
Comment (NicolasMahe) [reply cosmos/appfactory.go]: let's require something like 1wei in MESG token

File: cosmos/client.go (hunk 2 above)
Comment (NicolasMahe) [reply cosmos/client.go]: the adjusted gas seems really off compared to the actual used one.

File: cosmos/txbuilder.go (hunk 5 above)
Comment (krhubert) [reply cosmos/txbuilder.go]: This requires some refactoring, just to not add gasPrice as argument everywhere to already "argument-bolted" methods. Leaving decision to you.

**Pull request merged**